RAG/chat_interface.py

```diff
@@ -0,0 +1,140 @@
+from cohere import Client
+import os
+from langchain.llms import OpenAI
+from langchain.chains.question_answering import load_qa_chain
+import uuid
+import argparse
+from chroma_embedding import ChromaEmbedding
+from index_embedding import IndexEmbedding
+
+class Cohere():
+    """
+    A class that integrates with the Cohere API for conversational AI purposes.
+
+    Attributes:
+        __cohere_key (str): API key for Cohere.
+        __co (Client): Cohere client initialized with the API key.
+        __conversation_id (str): Unique ID for tracking the conversation.
+        __embedding Union(IndexEmbedding, ChromaEmbedding): an embedding system for the RAG to use
+        __max_tokens int: max number of tokns a response to a query can be
+    """
+    __cohere_key = os.getenv('COHERE_API_KEY')
+    __co = Client(__cohere_key)
+    __conversation_id = str(uuid.uuid4())
+
+    def __init__(self, chroma_embedding=True, use_openai=False, chunking_max_tokens=100, num_matches=5, max_tokens=500, dataset_path="RAG/datasets/"):
+        """
+        Initializes the Cohere class, setting up the embedding model used for queries.
+
+        Args:
+            chroma_embedding (bool): Flag to decide between using ChromaEmbedding or IndexEmbedding.
+            use_openai (bool): Determines if OpenAI embeddings should be used vs huggingface.
+            chunking_max_tokens (int): Maximum number of tokens for chunking in IndexEmbedding.
+            num_matches (int): Number of matching documents to return upon a query.
+            dataset_path (str [Path]): Path to the dataset directory.
+        """
+        self.__max_tokens = max_tokens
+        self.__embedding = None
+        if chroma_embedding:
+            self.__embedding = ChromaEmbedding(
+                 use_openai=use_openai,
+                 num_matches=num_matches,
+                 dataset_path=dataset_path
+            )
+        else:
+            self.__embedding = IndexEmbedding(
+                 use_openai=use_openai,
+                 chunking_max_tokens=chunking_max_tokens,
+                 num_matches=num_matches,
+                 dataset_path=dataset_path
+            )
+
+    def query(self, query):

```

Comment:
Also specifying return types would be good. Perhaps we can consider using Pydantic for more complex classes?